pkg/virt-operator/kubevirt.go

```diff
@@ -424,64 +426,94 @@ func (c *KubeVirtController) syncDeployment(kv *v1.KubeVirt) error {
+
+		// check if there is already an active KubeVirt deployment
+		// TODO move this into a new validating webhook
+		kvs := c.kubeVirtInformer.GetStore().List()
+		for _, obj := range kvs {
+			if fromStore, ok := obj.(v1.KubeVirt); ok {
+				if fromStore.UID == kv.UID {
+					continue
+				}
+				if isKubeVirtActive(&fromStore) {
+					logger.Warningf("There is already a KubeVirt deployment!")
+					util.UpdateCondition(kv, v1.KubeVirtConditionSynchronized, k8sv1.ConditionFalse, ConditionReasonDeploymentFailedExisting, "There is an active KubeVirt deployment")
+					return nil
+				}
 			}
 		}
-	}
 
-	// add finalizer to prevent deletion of CR before KubeVirt was undeployed
-	util.AddFinalizer(kv)
+		// add finalizer to prevent deletion of CR before KubeVirt was undeployed
+		util.AddFinalizer(kv)
 
-	// deploy
-	objectsAdded, err := creation.Create(kv, c.config, c.stores, c.clientset, &c.kubeVirtExpectations)
+		// deploy
+		objectsAdded, err := creation.Create(kv, c.config, c.stores, c.clientset, &c.kubeVirtExpectations)
 
-	if err != nil {
-		// deployment failed
-		util.UpdateCondition(kv, v1.KubeVirtConditionSynchronized, k8sv1.ConditionFalse, ConditionReasonDeploymentFailedError, fmt.Sprintf("An error occurred during deployment: %v", err))
+		if err != nil {
+			// deployment failed
+			util.UpdateCondition(kv, v1.KubeVirtConditionSynchronized, k8sv1.ConditionFalse, ConditionReasonDeploymentFailedError, fmt.Sprintf("An error occurred during deployment: %v", err))
 
-		logger.Errorf("Failed to create all resources: %v", err)
-		return err
+			logger.Errorf("Failed to create all resources: %v", err)
+			return err
+		}
+
+		if objectsAdded == 0 {
+			// add Created condition
+			util.UpdateCondition(kv, v1.KubeVirtConditionCreated, k8sv1.ConditionTrue, ConditionReasonDeploymentCreated, "All resources were created.")
+			logger.Info("All KubeVirt resources created")
+		}
 	}
 
-	if objectsAdded == 0 {
-		// deployment successful
+	// check if components are ready, but only if everything is created already
+	if util.HasCondition(kv, v1.KubeVirtConditionCreated) && c.isReady() {
+		logger.Info("All KubeVirt components ready")
 		kv.Status.Phase = v1.KubeVirtPhaseDeployed
-
-		// remove old conditions, add Created condition
-		kv.Status.Conditions = []v1.KubeVirtCondition{}
-		util.UpdateCondition(kv, v1.KubeVirtConditionCreated, k8sv1.ConditionTrue, ConditionReasonDeploymentCreated, "All resources were created.")
-
-		logger.Info("All KubeVirt resources created")
+		util.UpdateCondition(kv, v1.KubeVirtConditionReady, k8sv1.ConditionTrue, ConditionReasonDeploymentReady, "All components are ready.")
 		return nil
 	}
 
 	logger.Info("Processed deployment for this round")
 	return nil
 }
 
+func (c *KubeVirtController) isReady() bool {
+
+	for _, obj := range c.stores.DeploymentCache.List() {
+		if deployment, ok := obj.(*appsv1.Deployment); ok {
+			if deployment.Status.Replicas != deployment.Status.ReadyReplicas {
```

Comment:
Hey guys, chiming in here because I basically have the same ready condition in the cdi operator and I'm not sure it's correct.  Wondering if we should also ensure that `deployment.Spec.Replicas == deployment.Status.Replicas`?